client/src/ERC20Client.ts

```diff
@@ -0,0 +1,286 @@
+import { BigNumber, type BigNumberish } from '@ethersproject/bignumber';
+import * as blake from 'blakejs';
+import {
+  type CasperClient,
+  type CLKeyParameters,
+  type CLPublicKey,
+  type CLU256,
+  CLValueBuilder,
+  CLValueParsers,
+  Contracts,
+  type DeployUtil,
+  encodeBase16,
+  type Keys,
+  RuntimeArgs
+} from 'casper-js-sdk';
+
+const { Contract } = Contracts;
+
+export class ERC20Client extends Contract {
```

Comment:
I got your approach but we need to keep the API consistent with the [cep-78-js-client](https://github.com/casper-ecosystem/cep-78-enhanced-nft/blob/dev/client-js/src/index.ts#L77-L80)

Reply:
or we can use your approach and have `constructor` like this but add new method `init()` which will act as an CEP78 constructor, WDYT?

Reply:
https://github.com/casper-ecosystem/erc20/pull/55#discussion_r1119623968